astro/src/content/blog/grammarly-proves-ciam-not-optional.mdx

```diff
@@ -0,0 +1,56 @@
+
+By the way, these affected applications are not small time. From the article:
+
+* "Vidio is an online video streaming platform with 100M monthly active users."
+* "Bukalapak is one of the largest and most prominent eCommerce platforms in Indonesia with 150 million users."
+* Grammarly is probably the most widely known victim, with "30 million daily users".
+
+These are companies with millions of active users and hundreds or thousands of employees. These are not little startups in a garage. Yet for all three, "Login With Facebook" was insecurely implemented in such a way that account takeover was a real possibility.
+
+I'm not going to dig into the details in this post. [The article](https://salt.security/blog/oh-auth-abusing-oauth-to-take-over-millions-of-accounts) does a great job of that, including walking through how account takeover could be achieved.
+
+However, one lesson from the post is that whenever you are using OAuth tokens, you must always:
+
+* validate the signature of the token, preferably by using a library or introspection
+* validate the claims of the token
+
+This is true for any OAuth grant, including those covered by OIDC. If you want to learn more, [our article on secure signed JWTs](/articles/tokens/building-a-secure-jwt#consuming-a-jwt) goes into detail.
+
+![Verify those tokens.](/img/blogs/grammarly-oauth/verify-tokens-meme.png)
+
+## Securing Your Users Is Not Optional
+
+These breaches clearly demonstrate using proven customer identity and access management (CIAM) tools is non-optional today.
+
+Building on standards is great, but it isn't enough. A homegrown misimplementation of the standards has negative effects on your brand. Over [100 comments on HackerNews](https://news.ycombinator.com/item?id=38009291) about an article covering a security breach of yours is not great. But more importantly it impacts your users and their data. Again from the article:
+
+> "...the web editor [the compromised component] at Grammarly.com stores the data directly within Grammarly's account, which means an account takeover would give an attacker access to the victim’s stored documents. I myself am using Grammarly’s web editor for help as I've been writing this blog and writing emails, messages and documents in general."
+
+An account takeover at Grammarly, *thankfully entirely avoided per the security firm*, would have in some cases allowed an attacker to access users' private documents.
+
+Think of **your users' data being accessible** because of a mistake a developer made when implementing "Login With Facebook". I don't know about you, but this gives me the chills.
+
+CIAM is non-negotiable now. Users want to log in from multiple devices and access their profile data and all the functionality your applications offer easily and securely. They may be on a phone today and a computer tomorrow. They want to use Facebook to log in, or Google, or passkeys, or magic links. Any obstacle you put in front of them will frustrate them.
```

Comment:
If you're OK with it, I'd love it if you'd link to https://fusionauth.io/articles/identity-basics/magic-links

Reply:
Done.